Source/WebCore/platform/graphics/texmap/BitmapTexture.cpp

```diff
@@ -104,16 +104,22 @@ void BitmapTexture::reset(const IntSize& size, OptionSet<Flags> flags)
     }
 }
 
-void BitmapTexture::updateContents(const void* srcData, const IntRect& targetRect, const IntPoint& sourceOffset, int bytesPerLine)
+void BitmapTexture::updateContents(const void* srcData, const IntRect& targetRect, const IntPoint& sourceOffset, int bytesPerLine, PixelFormat pixelFormat)
 {
-    // We are updating a texture with format RGBA with content from a buffer that has BGRA format. Instead of turning BGRA
-    // into RGBA and then uploading it, we upload it as is. This causes the texture format to be RGBA but the content to be BGRA,
-    // so we mark the texture to convert the colors when painting the texture.
+    RELEASE_ASSERT(pixelFormat == PixelFormat::RGBA8 || pixelFormat == PixelFormat::BGRA8);
```

Comment:
The assert idea that I proposed here was to check that we weren't mixing formats. It can happen that a BGRA BitmapTexture gets updated with RGBA content and viceversa, and this is only allowed if we update the whole texture rectangle, cause we're moving all the pixels to the new format. Partial updates can't happen with different formats, and that's we should assert that is not happening.